src/main/java/com/example/petbuddybackend/utils/specification/CareSpecificationUtils.java

```diff
@@ -0,0 +1,187 @@
+        };
+    }
+
+    private static Specification<Care> maxCreatedTime(ZonedDateTime maxCreatedTime) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.lessThanOrEqualTo(root.get(SUBMITTED_AT), maxCreatedTime);
+        };
+    }
+
+    private static Specification<Care> minCareStart(LocalDate minCareStart) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.greaterThanOrEqualTo(root.get(CARE_START), minCareStart);
+        };
+    }
+
+    private static Specification<Care> maxCareStart(LocalDate maxCareStart) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.lessThanOrEqualTo(root.get(CARE_START), maxCareStart);
+        };
+    }
+
+    private static Specification<Care> minCareEnd(LocalDate minCareEnd) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.greaterThanOrEqualTo(root.get(CARE_END), minCareEnd);
+        };
+    }
+
+    private static Specification<Care> maxCareEnd(LocalDate maxCareEnd) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.lessThanOrEqualTo(root.get(CARE_END), maxCareEnd);
+        };
+    }
+
+    private static Specification<Care> minDailyPrice(BigDecimal minDailyPrice) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.greaterThanOrEqualTo(root.get(DAILY_PRICE), minDailyPrice);
+        };
+    }
+
+    private static Specification<Care> maxDailyPrice(BigDecimal maxDailyPrice) {
+        return (root, query, criteriaBuilder) -> {
+            return criteriaBuilder.lessThanOrEqualTo(root.get(DAILY_PRICE), maxDailyPrice);
+        };
+    }
+
+    public static Specification<Care> addClientEmailsFilter(Set<String> clientEmails) {
+        return (root, query, criteriaBuilder) -> {
+            Join<Care, Client> clientJoin = root.join(CLIENT);
+            return clientJoin.get(EMAIL).in(clientEmails);
+        };
+    }
+
+    public static Specification<Care> addCaretakerEmailsFilter(Set<String> caretakerEmails) {
+        return (root, query, criteriaBuilder) -> {
+            Join<Care, Caretaker> caretakerJoin = root.join(CARETAKER);
+            return caretakerJoin.get(EMAIL).in(caretakerEmails);
+        };
+    }
+
+    public static Specification<Care> addCaretakerEmailFilter(String caretakerEmail) {
+        return (root, query, criteriaBuilder) -> {
+            Join<Care, Caretaker> caretakerJoin = root.join(CARETAKER);
+            return criteriaBuilder.equal(caretakerJoin.get(EMAIL), caretakerEmail);
+        };
+    }
+
+    public static Specification<Care> addClientEmailFilter(String clientEmail) {
+        return (root, query, criteriaBuilder) -> {
+            Join<Care, Client> clientJoin = root.join(CLIENT);
+            return criteriaBuilder.equal(clientJoin.get(EMAIL), clientEmail);
+        };
+    }
```

Comment:
Nieużywane nigdzie

Reply:
jeszcze :)
z rozbiegu dodalem to do nastepnego taska: Zwracanie opiek klienta, czyli to samo ale z perspektywy klienta